plugin/core/protocol.py

```diff
@@ -171,7 +176,30 @@ class SignatureHelpTriggerKind:
     'relatedInformation': List[DiagnosticRelatedInformation]
 }, total=False)
 
-CompletionItem = Dict[str, Any]
+TextEdit = TypedDict('TextEdit', {
+    'range': RangeLsp,
+    'newText': str
+}, total=True)
+
+CompletionItem = TypedDict('CompletionItem', {
+    'label': str,
+    'kind': int,
+    'tags': List[CompletionItemTag],
+    'detail': str,
+    'documentation': Union[str, Dict[str, str]],
+    'deprecated': bool,
+    'preselect': bool,
+    'sortText': str,
+    'filterText': str,
+    'insertText': str,
+    'insertTextFormat': InsertTextFormat,
+    'insertTextMode': InsertTextMode,
+    'textEdit': TextEdit,
+    'additionalTextEdits': List[TextEdit],
+    'commitCharacters': str,
+    'command': Command,
+    'data': Any
```

Comment:
Personally, I would favor alphabetical order but you don't have to follow my suggestion (unless @rwols thinks the same maybe).